api-matlab/wrapper_physiobatch.m

```diff
@@ -0,0 +1,111 @@
+        [timing, ~] = spm_select(1,'.*.log','Specify scan timing file (..._Info.log)',{}, default_path);
+        % timing = input('Specify file path to the scan timing file (in template ending with _Info.log)', 's');
+        matlabbatch{1}.spm.tools.physio.log_files.cardiac = {cardiac};
+        matlabbatch{1}.spm.tools.physio.log_files.respiration = {respiration};
+        matlabbatch{1}.spm.tools.physio.log_files.scan_timing = {timing};
+    else
+        ME = MException('error:wronginput', 'Wrong vendor specified. Please specify one of BIDS, Philips, Biopac_Txt, GE, Siemens, Siemens_HCP, Siemens_Tics');
+        throw(ME);
+    end
+    
+    % read scan parameters
+    input_params = inputdlg({'Number of scans:','Number of slices:','Number of dummies:', 'TR:', 'Onset slice:'}, 'Parameters for physio');
+    
+    % scan parameters
+    matlabbatch{1}.spm.tools.physio.log_files.sampling_interval = [];
+    matlabbatch{1}.spm.tools.physio.log_files.relative_start_acquisition = 0;
+    matlabbatch{1}.spm.tools.physio.log_files.align_scan = 'last'; % sometimes first, sometimes last
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nslices = str2double(input_params{2});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.NslicesPerBeat = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.TR = str2double(input_params{4});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Ndummies = str2double(input_params{3});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nscans = str2double(input_params{1});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.onset_slice = str2double(input_params{5});
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.time_slice_to_slice = [];
+
+    % template
+    matlabbatch{1}.spm.tools.physio.scan_timing.sqpar.Nprep = [];
+    matlabbatch{1}.spm.tools.physio.scan_timing.sync.nominal = struct([]);
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.modality = 'ECG';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.min = 0.4;
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.initial_cpulse_select.auto_matched.file = 'initial_cpulse_kRpeakfile.mat';
+    matlabbatch{1}.spm.tools.physio.preproc.cardiac.posthoc_cpulse_select.off = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.output_multiple_regressors = 'multiple_regressors.txt';
+    matlabbatch{1}.spm.tools.physio.model.output_physio = 'physio.mat';
+    matlabbatch{1}.spm.tools.physio.model.orthogonalise = 'none';
+    matlabbatch{1}.spm.tools.physio.model.censor_unreliable_recording_intervals = false;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.c = 3;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.r = 4;
+    matlabbatch{1}.spm.tools.physio.model.retroicor.yes.order.cr = 1;
+    matlabbatch{1}.spm.tools.physio.model.rvt.no = struct([]);
+    matlabbatch{1}.spm.tools.physio.model.hrv.no = struct([]);
```

Comment:
Is there a specific reason for excluding RVT and HRV regressors? Do you think it's sensible to give the user the choice of which regressors to include?

This would also depend on what we want to do with the regressors, if it's for offline correction or just for quality plotting and comparison purposes. I think eventually we want to do both, and currently I have to think a bit first before deciding which has priority. But worth thinking about how this would be handled programmatically already.

Reply:
In general, I took the template from physIO and added only the information that is required. The reason for not asking the users about more parameter was that I thought the more things have to be manually input, the more it just resembles spm. In explanation, you can also use physio with the SPM GUI and you have to specify all these parameters, so if we do the same, it will become redundant to include it in rtQC. Therefore my idea was to provide a simple version that only requires the mandatory input fields, and all other parameters are filled in with the defaults from the physIO template. 

Of course however it does make sense to include RVT and HRV regressors if this information is relevant for further processing with rtQC. What exactly do you mean by "worth thinking about how this would be handled programmatically already"?  How to get the user input for these parameters? Or how to implement offline correction / plotting with the regressors?

Reply:
Sorry I think my use of English was not great in that sentence. I meant that we should think how we set up functions and parameters such that we can use either defaults or user input to decide which regressors to generate.

Based on today's discussion I would say we add these options to the defaults for now, for e.g. default HRV could be `no`. Then at some point we should make it clear to the user that clicking the "Get PhysIO batch" button will generate a specific set of regressors based on defaults, and that they can change these defaults if required. I think a full explanation would fit better into the software documentation, and a very short info sentence would be useful in the app itself.

Reply:
I agree, I will add some message that the user can change the defaults if required.